subprojects/gradle/ci-logger/src/main/kotlin/com/avito/utils/logging/CILoggingDestination.kt

```diff
@@ -0,0 +1,53 @@
+package com.avito.utils.logging
+
+import java.io.File
+import java.io.Serializable
+
+interface CILoggingDestination : Serializable {
+
+    fun write(message: String, throwable: Throwable?)
+    fun child(tag: String): CILoggingDestination
+}
+
+internal object StdoutDestination : CILoggingDestination {
+
+    override fun write(message: String, throwable: Throwable?) {
+        println(message)
+        println(throwable)
+    }
+
+    override fun child(tag: String): CILoggingDestination = this
+}
+
+internal object OnlyMessageStdoutDestination : CILoggingDestination {
+
+    override fun write(message: String, throwable: Throwable?) {
+        println(message)
+        throwable?.message?.also { println(it) }
+    }
+
+    override fun child(tag: String): CILoggingDestination = this
+}
+
+internal class FileDestination(
+    private val file: File
+) : CILoggingDestination {
+
+    override fun write(message: String, throwable: Throwable?) {
+        file.parentFile?.mkdirs()
```

Comment:
I think it's worth to prepare files only one time and don't touch them at each log message.
They are idempotent, but anyway it's I/O.

Reply:
I thought about that too
But it will be behavior changer

Reply:
Don't see changes in behavior.
We can use lazy or similar implementation here.